class-vip-support-user.php

```diff
@@ -657,6 +720,21 @@ public function is_a8c_email( $email ) {
 		return false;
 	}
 
+	/**
+	 * Is the email provided a VIP Support email alias.
+	 *
+	 * VIP Support staff without a real a8c email addresses will receive
+	 * an email address like `vip-support+<username>@automattic.com`.
+	 *
+	 * @param string $email an email address to check.
+	 *
+	 * @return bool true if the string is a VIP support email alias.
+	 */
+	public function is_vip_support_email_alias( $email ) {
+		$email_pattern = str_replace( [ '.', '@' ], [ '\.', '\+(.*)@' ], self::VIP_SUPPORT_EMAIL_ADDRESS );
+		return (bool) preg_match( "/{$email_pattern}/i", $email );
```

Comment:
This feels a bit convoluted. Can we just hardcode the regex in a const?

Reply:
I too felt this was a bit unclear when I wrote it (would be frustrating to read at 2am), but given the comment block and `$email_pattern` var name I thought the obscurity of it _might_ be okay. My goal was to avoid hardcoding the pattern so there didn't need to be two places for just one email address to be changed, in addition to the changes that would be needed and added in the unit tests. Before committing, I had written it out in a bit alternative way which may be less convoluted and easier to read:

```php
public function is_vip_support_email_alias( $email ) {
	list( $username, $domain ) = explode( '@', self::VIP_SUPPORT_EMAIL_ADDRESS );
	$domain = str_replace( '.', '\.', $domain );
	return (bool) preg_match( "/{$username}\+(.*)@{$domain}/i", $email );
}
```

However, I'm good with putting it as a constant if you are, especially since the email will probably never change anyways! Commit https://github.com/Automattic/vip-support/pull/73/commits/9a654b1f032cb80c12b48b8bc736232a1dd10192 pushed using constant.